poetry/installation/authenticator.py

```diff
@@ -0,0 +1,133 @@
+from typing import TYPE_CHECKING
+
+from poetry.utils._compat import urlparse
+from poetry.utils.password_manager import PasswordManager
+
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Optional
+    from typing import Tuple
+
+    from clikit.api.io import IO
+    from requests import Request  # noqa
+    from requests import Response  # noqa
+    from requests import Session  # noqa
+
+    from poetry.config.config import Config
+
+
+class Authenticator(object):
+    def __init__(self, config, io):  # type: (Config, IO) -> None
+        self._config = config
+        self._io = io
+        self._session = None
+        self._credentials = {}
+        self._password_manager = PasswordManager(self._config)
+
+    @property
+    def session(self):  # type: () -> Session
+        from requests import Session  # noqa
+
+        if self._session is None:
+            self._session = Session()
+
+        return self._session
+
+    def request(self, method, url, **kwargs):  # type: (str, str, Any) -> Response
+        from requests import Request  # noqa
+        from requests.auth import HTTPBasicAuth
+
+        request = Request(method, url)
+
+        username, password = self._get_credentials_for_url(url)
+
+        if username is not None and password is not None:
+            request = HTTPBasicAuth(username, password)(request)
+
+        session = self.session
```

Comment:
```suggestion
```

Reply:
I'd prefer to leave it as-is since it avoids the cost, albeit small, of the property access.